### PR TITLE
Add GSoC 2026 blog post on single-channel cellfinder support

### PR DESCRIPTION
**Pull request by aymuos15**
Adds a blog post covering my GSoC 2026 internship work on `cellfinder`.

### CODE REVIEW

File: docs/source/blog/gsoc_2026_cellfinder.md [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+**Project**: BrainGlobe: expand `cellfinder` input support to 2.5D and single-channel data <br>
+**Mentors**: [Igor Tatarnikov](https://github.com/IgorTatarnikov), [Alessandro Felder](https://github.com/alessandrofelder), [Adam Tyson](https://github.com/adamltyson)
+
+---
+
+## Project Overview
+
+`cellfinder` detects and classifies labelled cells in whole-brain microscopy volumes. It expected two channels: a signal channel carrying the labelled cells, and a background channel of autofluorescence that the classifier compares against to tell real cells from artefacts.
+
+That second channel earns its place. The original `cellfinder` paper ([Tyson et al., 2021](https://doi.org/10.1371/journal.pcbi.1009074)) describes it as "a secondary 'autofluorescence' channel that does not contain target signals but provides anatomical outlines", acquired so that the network can "'learn' the difference between neuron-based signals (only present in the primary signal channel), and other non-neuronal sources of fluorescence (potentially present in both channels)".
```
Comment (adamltyson) [inline docs/source/blog/gsoc_2026_cellfinder.md]: It may not be, but the first sentence reads like an LLM.

File: docs/source/blog/gsoc_2026_cellfinder.md [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+:location: London, UK
+:category: Blog
+:language: English
+:image: 0
+
+# GSoC 2026: single-channel support in cellfinder
+
+## Introduction
+
+Hi, I'm [Soumya](https://github.com/aymuos15). I was one of the [Neuroinformatics Unit](https://github.com/neuroinformatics-unit) GSoC interns for the 2026 summer term, and I worked on broadening the range of inputs [`cellfinder`](https://github.com/brainglobe/cellfinder) accepts, in both channel count and dimensionality.
```
Comment (adamltyson) [inline docs/source/blog/gsoc_2026_cellfinder.md]: Could you spell out (and link to) GSoC? Most BrainGlobe readers won't know what it is.

File: docs/source/blog/gsoc_2026_cellfinder.md (hunk 1 above)
Comment (adamltyson) [reply docs/source/blog/gsoc_2026_cellfinder.md]: Also the "anatomical outlines" quote isn't relevant, that relates to registration.

File: docs/source/blog/gsoc_2026_cellfinder.md [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+
+That second channel earns its place. The original `cellfinder` paper ([Tyson et al., 2021](https://doi.org/10.1371/journal.pcbi.1009074)) describes it as "a secondary 'autofluorescence' channel that does not contain target signals but provides anatomical outlines", acquired so that the network can "'learn' the difference between neuron-based signals (only present in the primary signal channel), and other non-neuronal sources of fluorescence (potentially present in both channels)".
+
+It is not free, though. As the [feature request](https://github.com/brainglobe/cellfinder/issues/352) behind this work put it, it is not always practical to double the data collected, and for some fluorophores a channel with only autofluorescence can be hard to obtain at all, which leaves it carrying little information while still costing a full acquisition. Making the background optional was the deliverable I want to talk about here.
```
Comment (adamltyson) [inline docs/source/blog/gsoc_2026_cellfinder.md]: Also reads like an LLM

File: docs/source/blog/gsoc_2026_cellfinder.md [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+
+### What changed
+
+The model builder now takes its channel count from the data instead of assuming two, so `background_array` can be `None` all the way from the public API to the classifier. With no background given, `cellfinder` switches to a single-channel model automatically, and a channel mismatch raises a clear error. Every pre-trained model `cellfinder` shipped expected two channels, so this path also needed retraining.
```
Comment (adamltyson) [inline docs/source/blog/gsoc_2026_cellfinder.md]: There's no reference to what the "model builder" is.

File: docs/source/blog/gsoc_2026_cellfinder.md [hunk 5]
```diff
@@ -0,0 +1,94 @@
...
+    background_array=None,
+    voxel_sizes=(5, 2, 2),
+)
+```
+
+In napari, you leave the background image unselected.
+
+### How well does it work
+
+I trained a single-channel ResNet-50 on the same serial two-photon dataset used for the existing model and published it on the [Hugging Face Hub](https://huggingface.co/brainglobe/cellfinder_single_channel_default), so it downloads on first use like the others.
```
Comment (adamltyson) [inline docs/source/blog/gsoc_2026_cellfinder.md]:
```suggestion
I trained a single-channel ResNet-50 on the same serial two-photon dataset used for the existing model and published it on the [Hugging Face Hub](https://huggingface.co/brainglobe/cellfinder_single_channel_default), so it downloads automatically on first use like the others.
```

File: docs/source/blog/gsoc_2026_cellfinder.md [hunk 6]
```diff
@@ -0,0 +1,94 @@
...
+
+Headline accuracy figures flatter it, so it is worth being precise about the cost.
```
Comment (adamltyson) [inline docs/source/blog/gsoc_2026_cellfinder.md]: Again, this might not be, but this reads like an LLM

File: docs/source/blog/gsoc_2026_cellfinder.md [hunk 7]
```diff
@@ -0,0 +1,94 @@
...
+- Plus a sweep skewing the class balance of the training data, to separate "not enough data" from "weaker input".
+- At a fixed threshold the two models look nearly identical, but that is an artefact: they sit at different points on their curves.
+- The comparison that shows the difference is at a matched operating point, here a recall of 0.99.
+
+| Training set | Threshold 1-ch | Threshold 2-ch | FP rate 1-ch (%) | FP rate 2-ch (%) |
+|---|---|---|---|---|
+| Balanced (47% cell) | 0.406 ± 0.018 | 0.495 ± 0.069 | 4.46 ± 0.23 | 3.58 ± 0.04 |
+| Skew 60% cell | 0.270 ± 0.046 | 0.599 ± 0.075 | 5.09 ± 0.32 | 3.83 ± 0.06 |
+| Skew 70% cell | 0.298 ± 0.082 | 0.663 ± 0.053 | 5.65 ± 0.39 | 4.18 ± 0.09 |
+| Skew 80% cell | 0.222 ± 0.044 | 0.760 ± 0.093 | 5.08 ± 0.21 | 4.05 ± 0.16 |
```
Comment (adamltyson) [inline docs/source/blog/gsoc_2026_cellfinder.md]: This section jumps straight into lots of technical points that won't be easily understood by others interested in cellfinder.

**Comment by aymuos15** [comment]
Thank you very much for the detailed feedback. I apologise it took so long to get back and it read poorly. I have tried to address all the comments and also added in a line about presenting at the summer school!

File: docs/source/blog/gsoc_2026_cellfinder.md [hunk 8]
```diff
@@ -0,0 +1,92 @@
...
+
+However, collecting a second channel doubles the data volume, and for some fluorophores it is difficult to obtain a channel containing only autofluorescence. These limitations are what originally motivated the [request for single-channel support](https://github.com/brainglobe/cellfinder/issues/352). Making this background channel optional was a major part of my project.
+
+### What changed
+
+The `build_model()` function creates the neural network used to classify candidate cells. The input shape passed to it now reflects the number of channels in the data, allowing the network to accept a signal channel alone. With no background given, `cellfinder` switches to a single-channel model automatically, and a channel mismatch raises a clear error. Every pre-trained model `cellfinder` shipped expected two channels, so this path also needed retraining.
```
Comment (alessandrofelder) [inline docs/source/blog/gsoc_2026_cellfinder.md]:
```suggestion
The `build_model()` function creates the neural network used to classify candidate cells. The input shape passed to it now reflects the number of channels in the data, allowing the network to accept a signal channel alone. With no background given, `cellfinder` switches to a newly trained single-channel default model automatically, and a channel mismatch raises a clear error. Previously, every pre-trained model `cellfinder` shipped expected two channels.
```

I don't think the word "path" is clear enough here, so I've made a suggestion for reformulation.

File: docs/source/blog/gsoc_2026_cellfinder.md [hunk 9]
```diff
@@ -0,0 +1,92 @@
...
+)
+```
+
+In napari, you leave the background image unselected.
+
+### How well does it work
+
+I used the same ResNet-50 architecture and training dataset as the standard two-channel model, but dropped the background channel during training to create our first single-channel model. I then uploaded it to the [Hugging Face Hub](https://huggingface.co/brainglobe/cellfinder_single_channel_default), marking the start of our transition to hosting models there. It downloads automatically on first use, just like the existing models.
+
+I then compared our results against the two-channel model to understand the trade-offs it would have. Early tests on data from a different microscope produced more false positives: objects incorrectly identified as cells. I tested both models on the same set of 10,756 image cubes, containing 5,066 cells and 5,690 non-cell objects, and varied the proportion of cells in the training data to see how it affected their performance.
```
Comment (alessandrofelder) [inline docs/source/blog/gsoc_2026_cellfinder.md]:
```suggestion
I then compared our results against the two-channel model to understand the trade-offs it would have. Early tests on data from a different microscope produced more false positives: objects incorrectly identified as cells. I tested both models on the same set of 10,756 image cubes, containing 5,066 cells and 5,690 non-cell objects, and varied the proportion of cells in the training data to see how it affected the models' performance.
```
(just because "their" could refer to image cubes, cells or models at this point in the sentence?)

File: docs/source/blog/gsoc_2026_cellfinder.md [hunk 10]
```diff
@@ -0,0 +1,92 @@
...
+| 47% (original balance) | 4.46% | 3.58% |
+| 60% | 5.09% | 3.83% |
+| 70% | 5.65% | 4.18% |
+| 80% | 5.08% | 4.05% |
+
+*Values are averages across three training runs.*
+
+### Closing the gap
+
+There is more work to be done here. Warm-start fine-tuning recovers roughly 40% of the gap and a top-hat variant around 28%, while distillation does nothing and hard-negative mining hurts. Substituting a blurred copy of the signal for the missing background also proved unsuccessful, since a real background channel is anti-correlated with the signal at a cell, with a mean background-to-signal intensity ratio of 0.05 at a true cell against 2.00 at a true non-cell, whereas a blurred copy is positively correlated by construction.
```
Comment (alessandrofelder) [inline docs/source/blog/gsoc_2026_cellfinder.md]:
```suggestion
I tried a few strategies to improve the default single-channel model's performance. Warm-start fine-tuning recovers roughly 40% of the gap and a top-hat variant around 28%, while distillation does nothing and hard-negative mining hurts. Substituting a blurred copy of the signal for the missing background also proved unsuccessful, since a real background channel is anti-correlated with the signal at a cell, with a mean background-to-signal intensity ratio of 0.05 at a true cell against 2.00 at a true non-cell, whereas a blurred copy is positively correlated by construction.
```

I think these improvement strategies are inaccessible to BrainGlobe users (and to me 😂 ): maybe this suggested reformulation helps provide some context.

I wonder whether we could also link to Wikipedia pages or good blogs about each improvement strategy to help provide context?

File: docs/source/blog/gsoc_2026_cellfinder.md [hunk 11]
```diff
@@ -0,0 +1,92 @@
...
+
+A performance drop from two channels was always expected. However, now, if you only have one channel, `cellfinder` runs. We have an explicit offering for it.
```
Comment (alessandrofelder) [inline docs/source/blog/gsoc_2026_cellfinder.md]:
```suggestion
A performance drop from two channels was always expected. However, now, if you only have one channel, you can know use `cellfinder`.
```

Don't think the last sentence adds anything?

**Review by alessandrofelder — COMMENTED** [review]
Thanks @aymuos15 - couple of wording suggestions from my side. Happy for you to push back if you disagree :) then we can merge!

File: docs/source/blog/gsoc_2026_cellfinder.md (hunk 10 above)
Comment (aymuos15) [reply docs/source/blog/gsoc_2026_cellfinder.md]: Shall I make an issue for this? I realise now that this code is only with me haha. Is there a sister repo somehwere for experimetns?

**Comment by aymuos15** [comment]
Thank you so much for all the suggestions! I will apply the last remaining one too once we decide on the issue thing

File: docs/source/blog/gsoc_2026_cellfinder.md (hunk 10 above)
Comment (alessandrofelder) [reply docs/source/blog/gsoc_2026_cellfinder.md]: Sorry I meant inaccessible in terms of language/meaning - not in terms of the code being available.

You can make a new issue here and add experimental benchmarking code in the benchmarks folder here?

Comment (aymuos15) [reply docs/source/blog/gsoc_2026_cellfinder.md]: Ah okay! Maybe we can discuss that tomorrow then, thank you!

File: docs/source/blog/gsoc_2026_cellfinder.md [hunk 12]
```diff
@@ -64,7 +64,7 @@ The single-channel model made more false-positive errors in every training setup
 
 ### Closing the gap
 
-There is more work to be done here. Warm-start fine-tuning recovers roughly 40% of the gap and a top-hat variant around 28%, while distillation does nothing and hard-negative mining hurts. Substituting a blurred copy of the signal for the missing background also proved unsuccessful, since a real background channel is anti-correlated with the signal at a cell, with a mean background-to-signal intensity ratio of 0.05 at a true cell against 2.00 at a true non-cell, whereas a blurred copy is positively correlated by construction.
+I tried a few strategies to improve the default single-channel model's performance. Warm-start fine-tuning recovers roughly 40% of the gap and a top-hat variant around 28%, while distillation does nothing and hard-negative mining hurts. Substituting a blurred copy of the signal for the missing background also proved unsuccessful, since a real background channel is anti-correlated with the signal at a cell, with a mean background-to-signal intensity ratio of 0.05 at a true cell against 2.00 at a true non-cell, whereas a blurred copy is positively correlated by construction.
```
Comment (alessandrofelder) [inline docs/source/blog/gsoc_2026_cellfinder.md]: Is it worth linking to 
* https://www.emergentmind.com/topics/warm-start-instruction-tuning
* https://huggingface.co/blog/Kseniase/kd
* https://en.wikipedia.org/wiki/Hard_negative_mining

(Just what I found with google search and thought looked reasonable... maybe you can eyeball the quality or have better references?)

Comment (aymuos15) [reply docs/source/blog/gsoc_2026_cellfinder.md]: I have added the links now, thank you!

**Review by alessandrofelder — APPROVED** [review]
